config/prometheus.spc

```diff
@@ -12,4 +12,9 @@ connection "prometheus" {
   #   - "prometheus_http_request.*" matches metrics starting with "prometheus_http_request"
   #   - ".*error.*" matches metrics containing the word "error"
   # metrics = [".+"]
+
+  # Header name and values to be used on the requests, generally for authentication.
```

Comment:
How would a user know which header name and value to use? Are there any docs from Prometheus on setting these values, along with standard or recommended values from them?

Reply:
Generally, prometheus only supports basic auth, I can add an example of that, but it will be a bit verbose, since it'll have to explain how to do base64 encoding. It's very late here so I'll do that tomorrow.

Reply:
@gcbirzan-plutoflume For your use case, would you use `header_name = "Authorization"` and `header_value = "Bearer Token Value"`, or would you use a different header name?

I was looking at Prometheus docs to learn more, and came across this doc, https://prometheus.io/docs/prometheus/1.8/configuration/configuration/, which mentions support for `bearer_token` and `bearer_token_file`. Are these config args related to this feature request?

Also, while trying to learn more, I found an [issue](https://github.com/prometheus/prometheus/issues/1724) in the Prometheus repo, which asks for custom header support, which doesn't affect usage of this plugin necessarily, but figured I'd mention in case the idea of custom headers is related.

If custom header names are needed, would you ever need to set more than one? To make this more flexible and not limit users to passing in 1 header name/value pairs, we could create a config arg as a map, e.g.,
```
headers = {
  "Authorization": "Bearer 42",
  "OtherHeader": "Foo"
}
```

If you only needed to set 1 header, then you'd only need one key/value pair in the map, but this would leave room for expansion.

Apologies if I got anything incorrect above, please let me know if you have any questions!